src/nginx/t/unregistered.t

```diff
@@ -183,26 +183,21 @@ sub verify_report {
   }
 }
 
-# 1st call: Check is called since api-key is provided
-# 2nd call: Check is NOT called since api-key is not provided
+# 1st call: Check is NOT called since api-key is not required even it is provided
+# 2nd call: Check is NOT called since api-key is not required nor provided
 # 3rd call: Check is called since api-key is provided
 # 4th call: Check is NOT called since api-key is NOT provided
 # Report are called for all of them.
 my @servicecontrol_requests = ApiManager::read_http_stream($t, 'servicecontrol.log');
-is(scalar @servicecontrol_requests, 6, 'Service control was called 6 times.');
+is(scalar @servicecontrol_requests, 5, 'Service control was called 5 times.');
 
-my ($check1, $report1, $report2, $check3, $report3, $report4) = @servicecontrol_requests;
+my ($report1, $report2, $check3, $report3, $report4) = @servicecontrol_requests;
 
 # /shelves?key=this-is-an-api-key-1
```

Comment:
to confirm: if there's no API Key, then Check isn't called?

Reply:
Yes